[docs] Add docs about span macros (#4918)
apollographql/router: docs/source/customizations/native.mdx

    # Any values here are passed to the plugin as part of your configuration
```

## Using macros

> To create custom metrics, traces, and spans you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate events and logs.

Reviewer: ```suggestion
To create custom metrics, traces, and spans, you can use [`tracing` macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate events and logs.
```

### Add custom metrics
<Note>

Make sure to [enable prometheus metrics](../configuration/telemetry/exporters/metrics/prometheus) in your configuration if you want to have metrics generated by the Router.

</Note>

> To create your custom metrics in [Prometheus](https://prometheus.io/) you can use the [event macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event.

Reviewer: ```suggestion
To create your custom metrics in [Prometheus](https://prometheus.io/) you can use [event macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate an event.
```

);
```

### Add custom spans
<Note>

> Make sure to [enable OpenTelemtry tracing](../configuration/telemetry/exporters/tracing/overview) in your configuration if you want to custom traces generated and linked by the Router.

Reviewer: ```suggestion
Make sure to [enable OpenTelemetry tracing](../configuration/telemetry/exporters/tracing/overview) in your configuration if you want customize the traces generated and linked by the router.
```

</Note>

> To create custom spans and traces you can use the [tracing macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate a span.

Reviewer: ```suggestion
To create custom spans and traces you can use [`tracing` macros](https://docs.rs/tracing/latest/tracing/index.html#macros) to generate a span.
```

```rust
use tracing::{Level, span};

> span!(target: "app_spans", Level::TRACE, "my span");

Reviewer 2: It's probably best to use info_span! as the level directly corresponds to the log level which most people will set to info.
